packages/test/test-drivers/src/odspTestDriver.ts

```diff
@@ -116,14 +127,37 @@ export class OdspTestDriver implements ITestDriver {
         options: OdspResourceTokenFetchOptions,
         config: IOdspTestLoginInfo & IClientConfig,
     ) {
-        // This function can handle token request for any multiple sites. Where the test driver is for a specific site.
-        const tokens = await this.odspTokenManager.getOdspTokens(
-            new URL(options.siteUrl).hostname,
+        const hostname = new URL(options.siteUrl).hostname;
+        try{
+            // This function can handle token request for any multiple sites.
+            // Where the test driver is for a specific site.
+            const tokens = await this.odspTokenManager.getOdspTokens(
+                new URL(options.siteUrl).hostname,
+                config,
+                passwordTokenConfig(config.username, config.password),
+                options.refresh,
+            );
+            return tokens.accessToken;
+        }catch(ex) {
+            if(config.inAutomation === true) {
+                throw ex;
+            }
+        }
+        const browserTokens = await this.odspTokenManager.getOdspTokens(
+            hostname,
             config,
-            passwordTokenConfig(config.username, config.password),
-            options.refresh,
+            {
+                type: "browserLogin",
+                navigator: (openUrl)=>{
+                    // eslint-disable-next-line max-len
+                    console.log(`Open the following url in a new private browser window, and login with user: ${config.username}`);
+                    console.log(`"${openUrl}"`);
+                },
+            },
+            true,
+            true,
         );
-        return tokens.accessToken;
+        return browserTokens.accessToken;
```

Comment:
Does this actually work?  Don't you need an auth redirect service running at localhost?